specs/WebNotification.md

```diff
@@ -0,0 +1,610 @@
+  [propget] HRESULT Silent([out, retval] BOOL* silent);
+
+  /// Specifies the time at which a notification is created or applicable (past,
+  /// present, or future).
+  /// Returns `null` if the optional Notification property does not exist.
+  [propget] HRESULT Timestamp([out, retval] double* timestamp);
+
+  /// Specifies a vibration pattern for devices with vibration hardware to emit.
+  /// Returns `null` if the optional Notification property does not exist.
+  [propget] HRESULT Vibrate([out, retval] ICoreWebView2UnsignedLongCollection** vibrationPattern);
+}
+```
+
+```csharp
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2NotificationReceivedEventArgs;
+    runtimeclass CoreWebView2Notification;
+    struct CoreWebView2NotificationAction
+    {
+        String Action;
+        String title;
+        String Icon;
+
+    };
+    struct CoreWebView2NotificationAction
+    {
+        String Action;
+        String title;
+        String Icon;
+
+    };
+    runtimeclass CoreWebView2Profile
+    {
+        ...
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2Profile3")]
+        {
+            // ICoreWebView2Profile3 members
+            event Windows.Foundation.TypedEventHandler<CoreWebView2Profile, CoreWebView2NotificationReceivedEventArgs> NotificationReceived;
+        }
+        ...
+    }
+    runtimeclass CoreWebView2
+    {
+        ...
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2_17")]
+        {
+            // ICoreWebView2_17 members
+            event Windows.Foundation.TypedEventHandler<CoreWebView2, CoreWebView2NotificationReceivedEventArgs> NotificationReceived;
+
+
+
+        }
+        ...
+    }
+    runtimeclass CoreWebView2NotificationReceivedEventArgs
+    {
+        // ICoreWebView2NotificationReceivedEventArgs members
+        String Uri { get; };
+        CoreWebView2Notification Notification { get; };
+        Boolean Handled { get; set; };
+        Windows.Foundation.Deferral GetDeferral();
+    }
+
+    runtimeclass CoreWebView2Notification
+    {
+        // ICoreWebView2Notification members
+        String Body { get; };
+        CoreWebView2NotificationDirectionKinds Direction { get; };
+        String Language { get; };
+        String Tag { get; };
+        String Icon { get; };
+        String Title { get; };
+        IVectorView<CoreWebView2NotificationAction> Actions { get; };
+        String Badge { get; };
+        String Image { get; };
+        Boolean Renotify { get; };
+        Boolean RequireInteraction { get; };
+        Boolean Silent { get; };
+        Double Timestamp { get; };
+        IVectorView<UInt64> Vibrate { get; };
+        // TODO: What should the proper data type be for 
+        // CoreWebView2Notification.Data? We use IDataObject for COM/C++.
+        // Data { get; };
+
+        event Windows.Foundation.TypedEventHandler<CoreWebView2Notification, Object> Closed;
+
+        void Show();
```

Comment:
All of these Report... methods occur under the covers asynchronously right? Do these need to be async methods or does it not matter for some reason?